### PR TITLE
Add / update ADOT e2e test cases

### PR DESCRIPTION
**Pull request by ivyostosh**
Added test cases for Cloudstack, Nutanix, and Tinkerbell; updated test cases for Docker and vSphere

*Issue #, if available:* https://github.com/aws/eks-anywhere-packages/issues/631

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/e2e/adot_test.go [hunk 1]
```diff
@@ -53,10 +53,58 @@ func TestCPackagesAdotVSphereKubernetes123UbuntuUpdateFlow(t *testing.T) {
...
+		framework.WithControlPlaneHardware(1),
+		framework.WithPackageConfig(t, packageBundleURI(v1alpha1.Kube122),
+			EksaPackageControllerHelmChartName, EksaPackageControllerHelmURI,
+			EksaPackageControllerHelmVersion, EksaPackageControllerHelmValues),
+	)
+	runCuratedPackagesAdotInstallTinkerbellSingleNodeFlow(test)
+}
+
+func TestCPackagesAdotTinkerbellBottleRocketKubernetes123SingleNodeFlow(t *testing.T) {
+	test := framework.NewClusterE2ETest(t,
```
Comment (junshun) [inline test/e2e/adot_test.go:93]: Should have credential check here

File: test/e2e/adot_test.go [hunk 2]
```diff
@@ -53,10 +53,58 @@ func TestCPackagesAdotVSphereKubernetes123UbuntuUpdateFlow(t *testing.T) {
...
+		framework.WithClusterFiller(api.WithKubernetesVersion(v1alpha1.Kube122)),
+		framework.WithPackageConfig(t, packageBundleURI(v1alpha1.Kube122),
+			EksaPackageControllerHelmChartName, EksaPackageControllerHelmURI,
+			EksaPackageControllerHelmVersion, EksaPackageControllerHelmValues),
+	)
+	runCuratedPackagesAdotInstallUpdateFlow(test)
+}
+
+func TestCPackagesAdotTinkerbellUbuntuKubernetes122SingleNodeFlow(t *testing.T) {
+	test := framework.NewClusterE2ETest(t,
```
Comment (junshun) [inline test/e2e/adot_test.go:81]: credential check here

Comment (junshun) [reply test/e2e/adot_test.go:81]:
```suggestion
        framework.CheckCuratedPackagesCredentials(t)
	test := framework.NewClusterE2ETest(t,
```

File: test/e2e/adot_test.go (hunk 1 above)
Comment (junshun) [reply test/e2e/adot_test.go:93]:
```suggestion
        framework.CheckCuratedPackagesCredentials(t)
	test := framework.NewClusterE2ETest(t,
```

File: test/e2e/adot_test.go [hunk 3]
```diff
@@ -66,11 +114,23 @@ func runCuratedPackagesAdotInstallSimpleFlow(test *framework.ClusterE2ETest) {
...
+		test.InstallCuratedPackage(adotPackageName, adotPackagePrefix+"-"+adotPackageName,
 			kubeconfig.FromClusterName(test.ClusterName), adotTargetNamespace,
 			"--set mode=deployment")
 		test.VerifyAdotPackageInstalled(adotPackagePrefix+"-"+adotPackageName, adotTargetNamespace)
 		test.VerifyAdotPackageDeploymentUpdated(adotPackagePrefix+"-"+adotPackageName, adotTargetNamespace)
 		test.VerifyAdotPackageDaemonSetUpdated(adotPackagePrefix+"-"+adotPackageName, adotTargetNamespace)
 	})
 }
+
+func runCuratedPackagesAdotInstallTinkerbellSingleNodeFlow(test *framework.ClusterE2ETest) {
```
Comment (junshun) [inline test/e2e/adot_test.go:126]: Should this type of function go in cluster.go? It seems to be mostly generic function calls more slated torwards tinkerbell with the only specific adot related to call a function test. Maybe having a tinkerbell setup and tinkerbell teardown is better. Especially if we want to test other flows

Comment (ivyostosh) [reply test/e2e/adot_test.go:128]: Please see https://github.com/aws/eks-anywhere/issues/4336

File: test/e2e/adot_test.go (hunk 1 above)
Comment (ivyostosh) [reply test/e2e/adot_test.go:95]: Updated

File: test/e2e/adot_test.go (hunk 2 above)
Comment (ivyostosh) [reply test/e2e/adot_test.go:82]: Updated

**Pull request merged**